client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -118,6 +122,48 @@ static Request index(IndexRequest indexRequest) {
         return new Request(method, endpoint, parameters.getParams(), entity);
     }
 
+    static Request update(UpdateRequest updateRequest) throws IOException {
```

Comment:
wondering if we should rather catch IOException and rethrow UncheckedIOException. getting tired of all these checked exceptions, which force us to move to CheckedFunction.